// Copyright 2022 The Chromium Authors
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

// If you change or add any fields in this file, update proto_visitors.h and
// potentially proto_enum_conversions.{h, cc}.

syntax = "proto2";

option java_multiple_files = true;
option java_package = "org.chromium.components.sync.protocol";

option optimize_for = LITE_RUNTIME;

package sync_pb;

// Autofill usage data related objects which contains usage data of a payment
// method related to a specific merchant.

message AutofillWalletUsageSpecifics {
  // A unique ID identifying this item. Used as the client tag.
  optional string guid = 1;

  // The type of usage data this entity is associated with.
  oneof usage_data_type {
    VirtualCardUsageData virtual_card_usage_data = 2;
  }

  message VirtualCardUsageData {
    // The instrument id of the actual card that the virtual card is related to.
    optional int64 instrument_id = 1;

    // The last four of the virtual card number. This is tied to the usage
    // data because the virtual card number may vary depending on merchants.
    optional string virtual_card_last_four = 2;

    // The url of the merchant where the virtual card is used.
    optional string merchant_url = 3;

    // The merchant the usage data is related to on the Android platform.
    optional string merchant_app_package = 4;
  }
}
